cvat/apps/dataset_manager/task.py

```diff
@@ -93,35 +94,47 @@ def add_prefetch_info(cls, queryset):
         ])
         label_qs = JobData.add_prefetch_info(label_qs)
 
+        task_data_queryset = models.Data.objects.all()
+        if prefetch_images:
+            task_data_queryset = task_data_queryset.select_related('video').prefetch_related(
+                Prefetch('images', queryset=models.Image.objects.order_by('frame'))
+            )
+
         return queryset.select_related(
             'segment',
             'segment__task',
```

Comment:
FYI, I don't think this impacts memory use heavily at the moment. It seems that using `select_related` results in different `Segment` and `Task` objects in Python, even if they are actually the same DB line. `prefetch_related`, however, results in the same objects with the same ids.